frontend/src/layouts/EventViewer/EventViewer.tsx

```diff
@@ -84,19 +123,31 @@ export const EventViewer = ({ events, selectedEventCallId, onEventClose, viewerP
 
       <div className="p-4 overflow-auto flex-1">
         <div className="font-mono select-text p-4 rounded-sm bg-muted overflow-x-auto w-full">
-          <ReactJson
-            // Needed to properly handle theme changes
-            key={theme.theme}
-            src={selectedEvent.details}
-            theme={viewerTheme}
-            iconStyle="square"
-            displayDataTypes={false}
-            displayObjectSize={false}
-            indentWidth={4}
-            enableClipboard={true}
-            style={{ fontFamily: "inherit", background: "none" }}
-            shouldCollapse={({ name }) => name !== "root"}
-          />
+          <div ref={jsonViewerRef}>
+            <Marker mark={searchValueMarker.searchValue}>
+              <ReactJson
+                // Needed to properly handle theme changes
+                key={theme.theme}
+                src={selectedEvent.details}
+                theme={viewerTheme}
+                iconStyle="square"
+                displayDataTypes={false}
+                displayObjectSize={false}
+                indentWidth={4}
+                enableClipboard={true}
+                style={{ fontFamily: "inherit", background: "none" }}
+                shouldCollapse={({ name, src }) => {
+                  if (searchBy === "eventName" || searchValue === "") {
+                    return name !== "root";
+                  }
```

Comment:
Isn't that reduntant? I think searchValue will be always an empty string here if `searchBy="eventName"` as you are always passing `searchValueByDetails`.
I would get rid of the searchBy prop altogether and only rely on searchValue